dashboard/src/actions/index.js

```diff
@@ -137,19 +139,18 @@ export const runQuery = query => {
                             // We display the response from server.
                         } else {
                             dispatch(addQuery(query));
-                            dispatch(
-                                saveSuccessResponse(JSON.stringify(result))
-                            );
+                            dispatch(saveSuccessResponse(null, result, true));
                         }
                     } else if (isNotEmpty(result)) {
                         dispatch(addQuery(query));
                         let mantainSortOrder = showTreeView(query);
-                        dispatch(saveSuccessResponse(null, result));
+                        dispatch(saveSuccessResponse(null, result, false));
                         dispatch(renderGraph(query, result, mantainSortOrder));
                     } else {
                         dispatch(
                             saveErrorResponse(
-                                "Your query did not return any results."
+                                "Your query did not return any results.",
+                                result
                             )
```

Comment:
is there a reason we are saving error messages in redux store? I think they might belong in react component states because they do not need to be shared application wide.